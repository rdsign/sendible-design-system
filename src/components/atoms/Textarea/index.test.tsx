import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Template } from '../../../../spec/template';
import { Textarea } from './index';

describe('textarea', () => {
    const onChange = jest.fn();

    const component = (
        <Template>
            <Textarea id="date" change={onChange} maxLength={50} />
        </Template>
    );

    test('should match snapshot', () => {
        const snapshot = create(component);
        expect(snapshot).toMatchSnapshot();
    });

    test('textarea should respect maxLenght', () => {
        render(component);
        const textarea = screen.getByRole('textbox');

        userEvent.type(textarea, 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.');
        expect(textarea).toHaveValue('Lorem ipsum dolor sit amet, consectetur adipiscing');
    });

    test('textarea should change its value on change', () => {
        render(
            <Template>
                <Textarea id="date" change={onChange} />
            </Template>,
        );
        const textarea = screen.getByRole('textbox');

        userEvent.type(textarea, 'My new value');
        expect(textarea).toHaveValue('My new value');
    });

    test('textarea should have initial value', () => {
        render(
            <Template>
                <Textarea id="date" change={onChange} initialValue="Hello World" />
            </Template>,
        );
        const textarea = screen.getByRole('textbox');

        expect(textarea).toHaveValue('Hello World');
    });
});
