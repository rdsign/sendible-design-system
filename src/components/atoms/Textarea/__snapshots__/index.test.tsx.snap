// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textarea should match snapshot 1`] = `
.c0 {
  background: #F4F4F4;
  border: 0.1rem solid #F4F4F4;
  border-radius: 0.4rem;
  color: #201E24;
  font-size: 2rem;
  height: 0px;
  overflow: hidden;
  padding: 1rem 2rem;
  resize: none;
}

.c0::-webkit-input-placeholder {
  color: #A6A5A7;
}

.c0::-moz-placeholder {
  color: #A6A5A7;
}

.c0:-ms-input-placeholder {
  color: #A6A5A7;
}

.c0::placeholder {
  color: #A6A5A7;
}

.c0:focus {
  background: #F6F5FF;
  border: 0.1rem solid #4D36FF;
}

<textarea
  className="c0"
  id="date"
  maxLength={50}
  onChange={[Function]}
/>
`;
